src/czbenchmarks/tasks/single_cell/perturbation_expression_prediction.py

```diff
@@ -0,0 +1,322 @@
+from typing import List, Dict, Literal
+import pandas as pd
+import numpy as np
+import logging
+from ..task import Task, TaskInput, TaskOutput
+from ...tasks.types import CellRepresentation
+from ...metrics import metrics_registry
+from ...metrics.types import MetricResult, MetricType
+from ..utils import binarize_values
+from ...constants import RANDOM_SEED
+from scipy import sparse as sp_sparse
+
+logger = logging.getLogger(__name__)
+
+
+class PerturbationExpressionPredictionTaskInput(TaskInput):
+    """Pydantic model for PerturbationTask inputs."""
+
+    de_results: pd.DataFrame
+    masked_adata_obs: pd.DataFrame
+    var_index: pd.Index
+    target_genes_to_save: Dict[str, List[str]]
+
+
+class PerturbationExpressionPredictionOutput(TaskOutput):
+    """Output for perturbation task."""
+
+    pred_log_fc_dict: Dict[str, np.ndarray]
+    true_log_fc_dict: Dict[str, np.ndarray]
+
+
+class PerturbationExpressionPredictionTask(Task):
+    display_name = "perturbation_expression_prediction"
+
+    def __init__(
+        self,
+        min_de_genes: int = 10,
+        control_gene: str = "non-targeting",
+        metric_column: str = "logfoldchange",
+        metric_type: str = "wilcoxon",
+        pval_threshold: float = 1e-4,
+        standardized_mean_diff: float = 0.5,
+        min_logfoldchange: float = 1.0,
```

Comment:
With the exception of `metric_column`, there are (now) all saved within the single cell perturbation dataset. To ensure they are always set the same, I think they should probably move to `PerturbationExpressionPredictionTaskInput` to indicate they should be set by the dataset. Does that seem like it helps to clarify where they originate from?